### PR TITLE
PSR-12 (hopefully relatively final) changes

### PR DESCRIPTION
**Pull request by michaelcullum**
From Korvin and my work together when we met up and worked on this in London.

### CODE REVIEW

**Comment by michaelcullum** [comment]
:+1:

**Comment by KorvinSzanto** [comment]
👍

File: proposed/extended-coding-style-guide.md [hunk 1]
```diff
@@ -145,6 +145,9 @@ When the opening `<?php` tag is on the first line of the file, it MUST be on its
 own line with no other statements unless it is a file containing markup outside of PHP
 opening and closing tags.
 
+Use declarations [Import statements] MUST never begin with a leading backslash as they
```
Comment (samdark) [inline proposed/extended-coding-style-guide.md]: Looks a bit weird to me. Either it should say "Use declarations" or "Import statements". Not both at the same time.

File: proposed/extended-coding-style-guide.md [hunk 2]
```diff
@@ -359,10 +363,29 @@ class ClassName
...
 
-### 4.3 Properties
+When using the `insteadof` and `as` operators they must be used as follows taking
+note of indentation, spacing and new lines.
+
+
+~~~php
+<?php
+
+class Talker {
```
Comment (samdark) [inline proposed/extended-coding-style-guide.md]: Shouldn't `{` of a class go to next line?

File: proposed/extended-coding-style-guide.md (hunk 1 above)
Comment (michaelcullum) [inline proposed/extended-coding-style-guide.md]: We use 'use declarations' throughout but I wasn't sure if we should clarify these are import statements. :/

File: proposed/extended-coding-style-guide.md (hunk 2 above)
Comment (michaelcullum) [inline proposed/extended-coding-style-guide.md]: Good spot. Could you PR please?

**Pull request merged**